gax-java/gax-httpjson/src/main/java/com/google/api/gax/httpjson/longrunning/stub/HttpJsonOperationsStub.java

```diff
@@ -309,22 +355,98 @@ protected HttpJsonOperationsStub(
     backgroundResources = new BackgroundResourceAggregation(clientContext.getBackgroundResources());
   }
 
-  private static <RequestT extends Message, ResponseT>
-      ApiMethodDescriptor<RequestT, ResponseT> getApiVersionedMethodDescriptor(
-          ApiMethodDescriptor<RequestT, ResponseT> methodDescriptor, String apiVersion) {
-    if (apiVersion == null) {
-      return methodDescriptor;
+  /* OperationsClient's RPCs are mapped to GET/POST/DELETE and this function only expects those HttpVerbs to be used */
+  private static String getValueBasedOnPatternCase(HttpRule httpRule) {
```

Comment:
The duplicate logic here and in the composer really bothers me, as the OperationClient should not exist in gax, but I guess we have to live with them for now unless we do a major refactoring.

Reply:
Agreed. I'm not sure if there is a cleaner/ better way. Maybe a static function in a util class for Operations that the generator can reference (?), but that seems like overkill.